js/interface.js

```diff
@@ -903,6 +928,112 @@ function showNothingFoundAlert(isShow) {
   }
 }
 
+function showAlertWithButtonGoToFolder(item, area) {
+  var goToButton = $('#alert-btn-action').attr('data-type', area.type);
+
+  if (area.type == 'organization') {
+    goToButton.attr('data-id', area.orgId);
+  } else if (area.type == 'app') {
+    goToButton.attr('data-id', area.appId);
+  } else {
+    goToButton.attr('data-id', area.id);
+  }
+
+  $('.alert-action').addClass('active');
+  $('.alert-message').text(item.length + ' item(s) moved');
+  setTimeout(function() {
+    setAlertVisibilityWhenMovingItems(false);
+  }, $goToFolderAlertTimeout);
+}
+
+// Check items if they are checked before moving
+function checkDraggedFileIfSelected(draggedItem, selectedItems) {
+  var res = false;
+  if (!selectedItems.length) {
+    return false;
+  }
+  for (var i = 0; i <= selectedItems.length; i++) {
+    if($(selectedItems[i]).attr('data-id') == draggedItem.id) {
+      res = true;
+      break;
+    }
+  }
+  return res;
+}
+
+// Change opacity when moving folders or files
+function setOpacityWhenMovingItems(item) {
+  $(item).css({ opacity: '0.3' }).removeClass('active');
+}
+
+// Create object for moving Items
+function moveItem(dropZone, isFolder) {
+  var movedPlace = {};
+  var parent = isFolder ? 'parentId' : 'mediaFolderId';
+  if (dropZone.type === 'organization') {
+    movedPlace.appId = null;
+    movedPlace[parent] = null;
+    movedPlace.organizationId = dropZone.orgId;
+  } else if (dropZone.type === 'app') {
+    movedPlace.appId = dropZone.appId;
+    movedPlace[parent] = null;
+  } else if (dropZone.fileType === 'folder') {
+    movedPlace[parent] = dropZone.id;
+  } else {
+    return false;
+  }
+  return movedPlace;
+}
+
+function moveItems(folderType, id, dropArea, element, items) {
+  if (folderType === 'folder') {
```

Comment:
no error caching for these API requests?

Reply:
fixed